server/users.js

```diff
@@ -21,31 +19,116 @@ module.exports = require('express').Router()
     // forbidden('listing users is not allowed'),
     (req, res, next) =>
     User.findAll()
-    .then(users => res.json(users))
+    .then(users => res.status(200).json(users))
+    .catch(next))
+  // query to see non-admin users
+  .get('/?isAdmin="false"', (req, res, next) =>
+    User.findAll({where: {isAdmin: req.query.isAdmin}})
+    .then(users => res.status(200).json(users))
+    .catch(next))
+  // query to see if user is admin
+  .get('/?isAdmin="true"', (req, res, next) =>
+    User.findAll({where: {isAdmin: req.query.isAdmin}})
+    .then(users => res.status(200).json(users))
+    .catch(next))
+  // admin or actual user can view
+  .get('/:id', (req, res, next) =>
+    User.findById(req.params.id, {
+      include: [Review]
+    })
+    .then(user => res.json(user))
+    .catch(next))
+  // cant create a new user with admin status
+  .post('/', (req, res, next) =>
+      User.create(req.body)
+      .then(user => res.status(201).json(user))
+      .catch(next))
+  .put('/:id', (req, res, next) =>
+    User.update(req.body, {where: {id: req.params.id}})
+    .then(([count, user]) => res.json(user))
+    .catch(next))
+  // admin or actual user can view
+  .delete('/:id', (req, res, next) =>
+    User.destroy({where: {id: req.params.id}})
+    .then(() => res.sendStatus(204)))
+
+  // REVIEWS ROUTE
+
+  .get('/:id/reviews', (req, res, next) =>
+    Review.findAll({where: {userId: req.params.reviews}})
+    .then(reviews => res.status(200).json(reviews))
     .catch(next))
 
-.get('/admin', (req, res, next) =>
-  User.findAll({
-    where: {
-      isAdmin: true
-    }
+  // ORDER ROUTES
+  // list of users orders
+  // specific user and admin access
+  .get('/:id/orders', (req, res, next) =>
+    Order.findAll({where: {userId: req.params.id}})
+    .then(orders => res.status(200).json(orders))
+    .catch(next))
+  // single user order with line items
+  // specific user and admin access
+  .get('/:id/orders/:orderId', (req, res, next) =>
+    Order.findById(req.params.orderId, {include: [LineItem]})
+    .then(order => res.status(200).json(order))
+    .catch(next))
+  // only single user can create
```

Comment:
Not sure what the single order part really means,I'll ask you the next time I see you, or you can slack me. Wouldn't admins want to create orders for users as well? I'm not sure if you have separate route or not. Just something to consider.